### PR TITLE
Update expect_column_values_to_be_of_type docs and tests (#659)

### PR DESCRIPTION
**Pull request by jcampbell**
Update and improve documentation for `expect_column_values_to_be_of_type` and `expect_column_values_to_be_in_type_list` (See #659 )

### CODE REVIEW

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_of_type.json [hunk 1]
```diff
@@ -274,7 +274,48 @@
             "success":true
           },
           "only_for": ["spark"]
+        },
+        {
+          "title": "dtype_object_and_type_object_still_has_aggregate_semantics",
```
Comment (abegong) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_of_type.json:279]: This is very thoughtful...

**Review by abegong — DISMISSED** [review]
This looks great. The improved tests are a nice bonus.

**Comment by jcampbell** [comment]
@abegong : I noticed and fixed a small typo ("unambiguous constraint" -> "unambiguous constraints"). New review should be a formality, then we can merge.

**Review by abegong — APPROVED** [review]
:+1:

**Pull request merged**